src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -124,6 +125,108 @@ def _ensure_monitor_with_config(
     return monitor
 
 
+def check_killswitch(
+    metric_kwargs: Dict,
+    project: Project,
+    monitor_slug: str,
+):
+    """
+    Enforce organization level monitor kll switch. Returns true if the
+    killswitch is enforced.
+    """
+    is_blocked = killswitch_matches_context(
+        "crons.organization.disable-check-in", {"organization_id": project.organization_id}
+    )
+    if is_blocked:
+        metrics.incr(
+            "monitors.checkin.dropped.blocked",
+            tags={**metric_kwargs},
+        )
+        logger.info(
+            "monitors.consumer.killswitch",
+            extra={"org_id": project.organization_id, "slug": monitor_slug},
+        )
+    return is_blocked
+
+
+def check_ratelimit(
+    metric_kwargs: Dict,
+    project: Project,
+    monitor_slug: str,
+    environment: str | None,
+):
+    """
+    Enforce check-in rate limits. Returns True if rate limit is enforced.
+    """
+    ratelimit_key = f"{project.organization_id}:{monitor_slug}:{environment}"
+
+    is_blocked = ratelimits.is_limited(
+        f"monitor-checkins:{ratelimit_key}",
+        limit=CHECKIN_QUOTA_LIMIT,
+        window=CHECKIN_QUOTA_WINDOW,
+    )
+
+    if is_blocked:
+        metrics.incr(
+            "monitors.checkin.dropped.ratelimited",
+            tags={**metric_kwargs},
+        )
+        logger.info(
+            "monitors.consumer.rate_limited",
+            extra={
+                "organization_id": project.organization_id,
+                "slug": monitor_slug,
+                "environment": environment,
+            },
+        )
+    return is_blocked
+
+
+def transform_checkin_uuid(
+    txn: Transaction | Span,
+    metric_kwargs: Dict,
+    monitor_slug: str,
+    check_in_id: str,
+):
+    """
+    Extracts the `UUID` object from the provided check_in_id. Failures will be logged.
+    Returns the UUID object and a boolean indicating if the provided GUID
+    signals usage of "the latest" check-in.
+    When the provided GUID is `0` use_latest_checkin will be True, indicating
+    that we should try and update the most recent check-in instead. A new UUID
+    will still be returned for use in the scanrio where there is no latest
+    check-in.
+    """
+    check_in_guid: uuid.UUID | None = None
+
+    try:
+        check_in_guid = uuid.UUID(check_in_id)
+    except ValueError:
+        metrics.incr(
+            "monitors.checkin.result",
+            tags={**metric_kwargs, "status": "failed_guid_validation"},
+        )
+        txn.set_tag("result", "failed_guid_validation")
+        logger.info(
+            "monitors.consumer.guid_validation_failed",
+            extra={"guid": check_in_id, "slug": monitor_slug},
+        )
+        return None, False
+
+    if check_in_guid is None:
+        return None, False
```

Comment:
do we need this?

Reply:
seems necessary so that in _process_message we can know whether to return early if `guid is None`

Reply:
wouldn't it throw an exception? like shouldn't we just catch all exceptions on line `211` and then remove this?

Reply:
ahhh i see yea tbh I also am unsure then, will leave for evan when he comes back don't wanna disturb the consumer too much right now

Reply:
This is to make the python type checker happy.